core/services/keeper/orm_test.go

```diff
@@ -0,0 +1,235 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/chainlink/core/internal/cltest"
+	"github.com/smartcontractkit/chainlink/core/services/keeper"
+	"github.com/smartcontractkit/chainlink/core/store"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var checkData = common.Hex2Bytes("ABC123")
+var executeGas = int32(10_000)
+
+func setupKeeperDB(t *testing.T) (*store.Store, keeper.KeeperORM, func()) {
+	store, cleanup := cltest.NewStore(t)
+	keeperORM := keeper.NewORM(store.DB)
+	return store, keeperORM, cleanup
+}
+
+func newUpkeep(registry keeper.Registry, upkeepID int64) keeper.UpkeepRegistration {
+	return keeper.UpkeepRegistration{
+		UpkeepID:   upkeepID,
+		ExecuteGas: executeGas,
+		Registry:   registry,
+		CheckData:  checkData,
+	}
+}
+
+func TestKeeperDB_Registries(t *testing.T) {
```

Comment:
Can we combine these tests into 1 or 2 which just does the CRUD to keep the db load lower (less setupKeepDB(t) calls) when running the test suite?

Reply:
The entire orm test suite is only 0.2 sec on my 4 core laptop. I'm inclined to keep the tests more isolated and simpler rather than bunching them all into one mega test function. I did add `t.Parallel()` to speed it up a bit though!

Reply:
I'm not worried about speed - just worried about like db connection pool exhaustion and db lock contention